### ISSUE TITLE
Correctly handle terminators in inline assembly

### ISSUE THREAD

**Issue by XAMPPRocky** [EmbarkStudios/rust-gpu#564]
Currently the following assembly code causes a `ConsumerError(MismatchedTerminator)` ICE.

### Code

```
// build-pass

use spirv_std as _;

#[spirv(any_hit_khr)]
pub fn main() {
    unsafe {
        asm!(r#"OpExtension "SPV_KHR_ray_tracing""#);
        asm!("OpCapability RayTracingKHR", "OpTerminateRayKHR");
    }
}
```

### Error

```
thread 'rustc' panicked at 'called `Result::unwrap()` on an `Err` value: ConsumerError(MismatchedTerminator)', crates/rustc_codegen_spirv/src/link.rs:423:58
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

error: internal compiler error: unexpected panic

note: the compiler unexpectedly panicked. this is a bug.

note: we would appreciate a bug report: https://github.com/EmbarkStudios/rust-gpu/issues/new

note: rustc 1.52.0-nightly (61edfd591 2021-03-20) running on x86_64-apple-darwin

note: compiler flags: -Z codegen-backend=/Users/erin.power/src/rust-gpu2/target/release/deps/librustc_codegen_spirv.dylib -Z unstable-options -Z crate-attr=no_std -Z crate-attr=feature(register_attr,asm) -Z crate-attr=register_attr(spirv) -C prefer-dynamic -C overflow-checks=off -C debug-assertions=off -C debuginfo=2 -C embed-bitcode=no --crate-type dylib
```

**Comment by XAMPPRocky** [comment]
Currently you can work around this by adding unused `OpLabel` at the end of the `asm!` block. 

```rust
asm!("OpKill", "%foo = OpLabel");
```

**Comment by khyperia** [comment]
Fixed by #717
